src/Builder/AsyncBuilderTrait.php

```diff
@@ -56,8 +57,9 @@ public function generateAsync(): void
         }
 
         if (null !== $this->fileName) {
-            // Gotenberg will add the extension to the file name (e.g. filename : "file.pdf" => generated file : "file.pdf.pdf").
-            $headers['Gotenberg-Output-Filename'] = $this->fileName;
+            /** @var string $basename Gotenberg will add the extension to the file name */
+            $basename = preg_replace('/\.[^.]+$/', '', $this->fileName);
```

Comment:
```suggestion
            $basename = preg_replace('/\.pdf$/', '', $this->fileName);
```

or something similar. But we shouldn't prevent from having a `.something` at the end. So either we detect that it ends with `.pdf`, `.jpg` or whatever and we raise a warning log. Or we simply drop this part and use the raw property value.

Reply:
IMO, we should only document that the extension will be added by Gotenberg. We should limit user input modification to avoid any misunderstanding.

Reply:
hmmm IMO limiting user modification is weirder. We shouldn't validate or modify whatever is input by the developper.

Reply:
I want to limit "input modification", not "limit input" :smile: